lib/Hakyll/Core/Runtime.hs

```diff
@@ -289,33 +290,32 @@ chase id' = do
                 }
 
         -- Try something else first
-        CompilerRequire dep c -> do
-            let (depId, depSnapshot) = dep
-            Logger.debug logger $
-                "Compiler requirement found for: " ++ show id' ++
-                ", requirement: " ++ show depId
-
+        CompilerRequire reqs c -> do
             let done      = runtimeDone state
                 snapshots = runtimeSnapshots state
                 deps      = runtimeDependencies state
 
-            -- Done if we either completed the entire item (runtimeDone) or
-            -- if we previously saved the snapshot (runtimeSnapshots).
-            let depDone =
-                    depId `S.member` done ||
-                    (depId, depSnapshot) `S.member` snapshots
+            deps' <- fmap join . for reqs $ \dep -> do
+                let (depId, depSnapshot) = dep
+                Logger.debug logger $
+                    "Compiler requirement found for: " ++ show id' ++
+                    ", requirement: " ++ show depId
+
+                -- Done if we either completed the entire item (runtimeDone) or
+                -- if we previously saved the snapshot (runtimeSnapshots).
+                let depDone =
+                        depId `S.member` done ||
+                        (depId, depSnapshot) `S.member` snapshots
+
+                Logger.debug logger $ "Require " ++ show depId ++
+                    " (snapshot " ++ depSnapshot ++ ") "
```

Comment:
I think we could merge the two log messages.

Reply:
I concur.

Reply:
Done.